test/e2e_flags.go

```diff
@@ -73,9 +82,47 @@ func initializeFlags() *EnvironmentFlags {
 
 	flag.StringVar(&f.Tag, "tag", "latest", "Provide the version tag for the test images.")
 
+	klog.InitFlags(nil)
+	flag.Set("v", "2") // A normal level of logging for testing
+	flag.Set("alsologtostderr", "true")
+
 	return &f
 }
 
+func printFlags() {
+	fmt.Print("Test Flags: {")
+	flag.CommandLine.VisitAll(func(f *flag.Flag) {
+		fmt.Printf("'%s': '%s', ", f.Name, f.Value.String())
+	})
+	fmt.Println("}")
+}
+
+var flagsSetup = false
+var flagsMutex = &sync.Mutex{}
```

Comment:
Are we doing this more than once? Should it be just `sync.Once`?

Reply:
I would like to verify the function doesn't get called during package initialization and I can't seem to do that with sync.Once() without introducing a second variable and thus having essentially the same code now + boilerplate func calls, so I'll use your other suggestions to make it cleaner.